test/test_grid_generators.jl

```diff
@@ -0,0 +1,82 @@
+using Test
+using Ferrite
+using Tensors
+
+# Helper function to test grid generation for a given floating-point type
+function test_generate_grid(T::Type)
+    # Helper function to check the type of nodes and cells in the grid
+    function check_grid_types(grid::Grid{dim, C, T}) where {dim, C <: Ferrite.AbstractCell, T}
+        # Check nodes
+        for node in grid.nodes
+            @test typeof(node) == Node{dim, T}
+            @test typeof(get_node_coordinate(node)) == Vec{dim, T}
+        end
+        
+        # Check cells
+        @test all(typeof(cell) == C for cell in grid.cells)
+    end
+    
+    # Testing Line grid
+    nel = (10,)
+    left = Vec{1,T}((-1.0,))
+    right = Vec{1,T}((1.0,))
+    grid = generate_grid(Line, nel, left, right)
+    @test isa(grid, Grid{1, Line, T})
```

Comment:
These tests are also nice to put into a loop over the cell types you want to test
```suggestion
    for CT in (Line, QuadraticLine, Quadrilateral, ...)
        rdim = Ferrite.getrefdim(CT)
        nels = ntuple(i -> 2, rdim)
        left = - ones(Vec{rdim,T}
        right =  ones(Vec{rdim,T}
        grid = generate_grid(CT, nels, left, right)
        @test isa(grid, Grid{rdim, CT, T})
    end
```